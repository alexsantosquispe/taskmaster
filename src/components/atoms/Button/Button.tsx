import cn from 'clsx';
import type { ReactNode } from 'react';
import { twMerge } from 'tailwind-merge';

interface ButtonProps {
  label: string;
  onClick: () => void;
  ariaLabel: string;
  icon?: ReactNode;
  isSecondary?: boolean;
  isDisable?: boolean;
  className?: string;
}

export const Button = ({
  label,
  onClick,
  ariaLabel,
  icon,
  isSecondary = false,
  isDisable = false,
  className
}: ButtonProps) => {
  return (
    <button
      aria-label={ariaLabel}
      className={twMerge(
        'flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out',
        cn({
          'bg-blue-800 text-white shadow-lg dark:bg-orange-700': !isSecondary,
          'hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80':
            !isSecondary && !isDisable,

          'text-primary dark:text-white': isSecondary,
          'hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15':
            isSecondary && !isDisable,

          'cursor-not-allowed': isDisable,
          'bg-primary/30 shadow-none dark:bg-white/15 dark:text-white/40':
            !isSecondary && isDisable,
          'text-primary/50 dark:text-white/40': isSecondary && isDisable
        }),
        className
      )}
      onClick={onClick}
    >
      {icon && <div>{icon}</div>}
      {label}
    </button>
  );
};
