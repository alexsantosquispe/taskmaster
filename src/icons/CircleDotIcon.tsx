import type { SvgIconProps } from '../models/types';
import { twMerge } from 'tailwind-merge';

export const CircleDotIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge('size-5', className)}
    >
      <circle cx="12" cy="12" r="10" />
      <circle cx="12" cy="12" r="1" />
    </svg>
  );
};
