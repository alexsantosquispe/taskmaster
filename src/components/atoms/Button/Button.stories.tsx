import type { Meta, StoryObj } from '@storybook/react';

import { WrapperUI } from '@/utils/test.utils';
import { CirclePlusIcon } from '../../../icons';
import { Button } from './Button';

const meta = {
  title: 'Atoms/Button',
  component: Button,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof Button>;

export const Default: Story = () => {
  return (
    <div className="grid grid-cols-4 gap-8">
      <WrapperUI>
        <Button
          label="Default Button"
          ariaLabel="default button"
          onClick={() => {}}
        />
      </WrapperUI>

      <WrapperUI>
        <Button
          label="Button Disabled"
          ariaLabel="default Button Disabled"
          onClick={() => {}}
          isDisable={true}
        />
      </WrapperUI>

      <WrapperUI>
        <Button
          label="Secondary Button"
          ariaLabel="secondary button"
          onClick={() => {}}
          isSecondary={true}
        />
      </WrapperUI>

      <WrapperUI>
        <Button
          label="Secondary Disabled"
          ariaLabel="secondary button disabled"
          onClick={() => {}}
          isSecondary={true}
          isDisable={true}
        />
      </WrapperUI>

      <WrapperUI>
        <Button
          label="Button with Icon"
          ariaLabel="default button"
          icon={<CirclePlusIcon />}
          onClick={() => {}}
        />
      </WrapperUI>

      <WrapperUI>
        <Button
          label="Secondary with Icon"
          ariaLabel="secondary button disabled"
          onClick={() => {}}
          icon={<CirclePlusIcon />}
          isSecondary={true}
        />
      </WrapperUI>
    </div>
  );
};

Default.args = {};
