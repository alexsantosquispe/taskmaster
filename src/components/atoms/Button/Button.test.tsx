import { render } from '@testing-library/react';
import { CirclePlusIcon } from '../../../icons';
import { Button } from './Button';

describe('Button', () => {
  const onClickMock = jest.fn();
  const props = {
    label: 'Test button',
    onClick: onClickMock,
    ariaLabel: 'test button'
  };

  it('should render the button correctly', () => {
    let component = render(<Button {...props} />);
    expect(component).toMatchSnapshot();

    component = render(<Button {...props} isDisable={true} />);
    expect(component).toMatchSnapshot();

    component = render(<Button {...props} isSecondary={true} />);
    expect(component).toMatchSnapshot();

    component = render(
      <Button {...props} isSecondary={true} isDisable={true} />
    );
    expect(component).toMatchSnapshot();

    component = render(<Button {...props} isLoading={true} />);
    expect(component).toMatchSnapshot();
  });

  it('should render the button with icon correctly', () => {
    let component = render(<Button {...props} icon={<CirclePlusIcon />} />);
    expect(component).toMatchSnapshot();

    component = render(
      <Button {...props} icon={<CirclePlusIcon />} isSecondary={true} />
    );
    expect(component).toMatchSnapshot();
  });
});
