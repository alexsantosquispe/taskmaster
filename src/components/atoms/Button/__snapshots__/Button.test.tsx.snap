// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button should render the button correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
      type="button"
    >
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-white cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-white cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
      disabled=""
      type="button"
    >
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button correctly 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-white cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
        type="button"
      >
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
      type="button"
    >
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button correctly 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-white cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out cursor-not-allowed text-neutral-400 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out cursor-not-allowed text-neutral-400 dark:text-white/40"
      disabled=""
      type="button"
    >
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button correctly 5`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-white cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out cursor-not-allowed text-neutral-400 dark:text-white/40"
        disabled=""
        type="button"
      >
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
        disabled=""
        type="button"
      >
        <div>
          <svg
            aria-hidden="true"
            class="h-4 w-4 animate-spin"
            fill="none"
            role="status"
            viewBox="0 0 100 101"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
              fill="white"
            />
            <path
              d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
              fill="currentColor"
            />
          </svg>
        </div>
        Loading...
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out cursor-not-allowed bg-neutral-300 shadow-none dark:bg-white/15 dark:text-white/40"
      disabled=""
      type="button"
    >
      <div>
        <svg
          aria-hidden="true"
          class="h-4 w-4 animate-spin"
          fill="none"
          role="status"
          viewBox="0 0 100 101"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
            fill="white"
          />
          <path
            d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
            fill="currentColor"
          />
        </svg>
      </div>
      Loading...
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button with icon correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M8 12h8"
            />
            <path
              d="M12 8v8"
            />
          </svg>
        </div>
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
      type="button"
    >
      <div>
        <svg
          class="size-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M8 12h8"
          />
          <path
            d="M12 8v8"
          />
        </svg>
      </div>
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button should render the button with icon correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out bg-blue-800 text-white shadow-lg dark:bg-orange-700 hover:cursor-pointer hover:bg-blue-800/80 dark:hover:bg-orange-800/80"
        type="button"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M8 12h8"
            />
            <path
              d="M12 8v8"
            />
          </svg>
        </div>
        Test button
      </button>
    </div>
    <div>
      <button
        aria-label="test button"
        class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
        type="button"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M8 12h8"
            />
            <path
              d="M12 8v8"
            />
          </svg>
        </div>
        Test button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="test button"
      class="flex w-fit items-center justify-center gap-2 rounded-lg px-3 py-2 text-sm font-semibold transition-colors duration-200 ease-in-out text-primary dark:text-white hover:text-accent dark:hover:text-accent-dark hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15"
      type="button"
    >
      <div>
        <svg
          class="size-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M8 12h8"
          />
          <path
            d="M12 8v8"
          />
        </svg>
      </div>
      Test button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
